### PR TITLE
Handling exception to call the hook in standard_ExecutorStart

### PR DESCRIPTION
**Pull request by adam8157**
Issue #12690, the code path in `standard_ExecutorStart` didn't handle
the exception of `Policy*AssignOperatorMemoryKB`, which may cause the OOM
exception not handled in `standard_ExecutorStart` but thrown to upper
`PortalStart` methods, in that case, the hook needed by GPCC will not
be called. This commit fixed the issue.

### CODE REVIEW

**Review by Aegeaner — APPROVED** [review]
LGTM.

File: src/backend/executor/execMain.c [hunk 1]
```diff
@@ -397,7 +399,17 @@ standard_ExecutorStart(QueryDesc *queryDesc, int eflags)
 				default:
 					Assert(IsResManagerMemoryPolicyNone());
 					break;
+				}
 			}
+			PG_CATCH();
+			{
+				/* GPDB hook for collecting query info */
+				if (query_info_collect_hook)
+					(*query_info_collect_hook)(QueryCancelCleanup ? METRICS_QUERY_CANCELED : METRICS_QUERY_ERROR, queryDesc);
```
Comment (adam8157) [inline src/backend/executor/execMain.c:408]: @violet2016 is calling `query_info_collect_hook` safe here? `queryDesc->estate` is NULL

Comment (yangru) [reply src/backend/executor/execMain.c:408]: There's no estate related info need to access when METRICS_QUERY_CANCELED or METRICS_QUERY_ERROR event sent to metrics_collector, and it won't call mppExecutorCleanup again. So I'm OK with this change.

Comment (violet2016) [reply src/backend/executor/execMain.c:408]: I agree with Ru. We only use queryDesc->estate when query submit, so this code is safe.

**Review by violet2016 — APPROVED** [review]

**Comment by violet2016** [comment]
GPCC needs this pr merged into master, thanks.

**Pull request merged**